editoast/src/models/infra.rs

```diff
@@ -150,23 +151,26 @@ impl Infra {
     }
 
     pub async fn clone(infra_id: i64, db_pool: Data<DbPool>, new_name: String) -> Result<Infra> {
-        let infra_to_clone = Infra::retrieve(db_pool.clone(), infra_id).await?.unwrap();
-
-        let mut conn = db_pool.get().await?;
-        sql_query(
-            "INSERT INTO infra (name, railjson_version, owner, version, generated_version, locked, created, modified
+        let infra_to_clone = Infra::retrieve(db_pool.clone(), infra_id).await?;
```

Comment:
To avoid having to add a level of indentation you can do like so:

```suggestion
        let Some(infra_to_clone) = Infra::retrieve(db_pool.clone(), infra_id).await? else {
            return Err(InfraError::NotFound(infra_id).into());
        }
```

Reply:
I actually prefer the @Tristramg suggestion :smile:.